compiler/rustc_mir_build/src/build/matches/mod.rs

```diff
@@ -1214,31 +1225,35 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
     /// This is the most subtle part of the matching algorithm. At
     /// this point, the input candidates have been fully simplified,
     /// and so we know that all remaining match-pairs require some
-    /// sort of test. To decide what test to do, we take the highest
+    /// sort of test. To decide what test to perform, we take the highest
     /// priority candidate (last one in the list) and extract the
```

Comment:
cc https://rust-lang.zulipchat.com/#narrow/stream/131828-t-compiler/topic/match.20candidate.20priority.20order

Reply:
I think this comment is wrong, as in the code itself we see the highest priority candidate being referred to as the first in the list:
https://github.com/rust-lang/rust/blob/2ea320aa346851638b6b5df38ab3795ec1b73eff/compiler/rustc_mir_build/src/build/matches/mod.rs#L1334-L1335
This also seems to match all other references to the sorting order.